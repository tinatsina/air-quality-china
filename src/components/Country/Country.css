:root {
  --dark-blue: #4369b2;
  --light-blue: #5485df;
  --night-blue: #35548c;
  --font-paint: #fff;
  --alt-blue: #3f63a6;
}

.aqi-navbar > div > a > svg {
  color: #fff;
  width: 1.5em;
  height: 1.5em;
  margin-top: 0.65em;
  margin-left: 1em;
}

.aqi-navbar > p {
  margin-top: 0.5em;
}

.aqi-navbar {
  display: flex;
}

.aqi-big-display {
  display: flex;
  flex-direction: row-reverse;
  color: var(--font-paint);
}

.aqi-big-display > div {
  width: 65%;
  text-align: right;
  margin-right: 5%;
}

.aqi-big-display > img {
  width: 30%;
  margin: 5%;
  opacity: 0.33;
}

.city-header {
  padding: 0;
  margin: 0;
  color: var(--font-paint);
}

.city-header > li {
  display: flex;
  background-color: var(--light-blue);
}

.city-header > li:nth-child(even) {
  background-color: var(--dark-blue);
}

.city-header > li > p:first-child {
  margin-right: auto;
  margin-left: 5%;
}

.city-header > li > p:last-child {
  margin-right: 5%;
}
