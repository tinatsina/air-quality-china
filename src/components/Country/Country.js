import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';

const Country = () => {
  // Get the state data for the webpage we have just loaded
  const e = useSelector((state) => state.aqiReducer);
  console.log(e);
  return (
    <>
      <h1>
        Hello
      </h1>
      <h2>AQI Data</h2>
      <Link to="/">Home Page</Link>
      <ul className="city-header">
        <li>{e.co}</li>
        <li>{e.h}</li>
        <li>{e.no2}</li>
        <li>{e.o3}</li>
        <li>{e.p}</li>
        <li>{e.pm10}</li>
        <li>{e.pm25}</li>
        <li>{e.so2}</li>
        <li>{e.t}</li>
        <li>{e.w}</li>
      </ul>
      <br />
    </>
  );
};

export default Country;
