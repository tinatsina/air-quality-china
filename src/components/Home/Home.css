:root {
  --dark-blue: #4369b2;
  --light-blue: #5485df;
  --night-blue: #35548c;
  --font-paint: #fff;
  --alt-blue: #3f63a6;
}

body {
  margin: auto;
  width: 60%;
}

nav {
  background-color: var(--dark-blue);
  color: var(--font-paint);
}

nav > p {
  margin: auto;
  line-height: 2em;
  height: 3em;
  text-align: center;
  font-size: xx-large;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

footer > p {
  margin: 0;
  text-align: center;
}

footer > p > a {
  color: var(--font-paint);
}

.china-display {
  display: flex;
  background-color: var(--light-blue);
  color: var(--font-paint);
}

.china-display > img {
  width: 75%;
  opacity: 0.25;
}

.china-display > p {
  font-size: larger;
  font-weight: 300;
}

.china-search-box {
  background-color: var(--light-blue);
  display: flex;
}

.china-search-box > input {
  border-radius: 1em;
  border: none;
  width: 60%;
  margin: 0 5%;
}

.china-search-box > button {
  width: 25%;
  height: 3em;
  border-radius: 20%;
  border: none;
}

.night-blue-list {
  background-color: var(--night-blue);
  color: var(--font-paint);
}

.night-blue-list > p {
  margin: 0;
}

.home-container {
  padding: 0;
  margin: 0;
  background-color: var(--dark-blue);
  color: var(--font-paint);
}

.home-card {
  display: inline-block;
  width: 50%;
  background-color: var(--night-blue);
}

.home-card:nth-child(1),
.home-card:nth-child(4),
.home-card:nth-child(5),
.home-card:nth-child(8),
.home-card:nth-child(9) {
  background-color: var(--light-blue);
}

.image-arrow-conatiner {
  display: flex;
}

.image-arrow-conatiner > img {
  width: 60%;
  margin: 10%;
  opacity: 0.25;
}

.image-arrow-conatiner > a > svg {
  width: 1.5em;
  height: 1.5em;
  color: var(--font-paint);
}

.image-arrow-conatiner > a {
  padding-top: 1em;
}

.city-name {
  font-size: x-large;
  font-weight: 400;
  margin: 0 0.25em;
  text-align: end;
}

.city-aqi {
  font-size: larger;
  font-weight: 300;
  margin: 0 0.25em;
  text-align: end;
}

.aqi-big-display {
  background-color: var(--light-blue);
  margin: 0;
}

.aqi-big-display > h2 {
  margin-bottom: 0;
  line-height: 3.5em;
  margin-top: 0;
}

.aqi-big-display > h3 {
  font-size: xx-large;
  margin: 0;
}

footer {
  background-color: var(--night-blue);
  color: var(--font-paint);
  height: 5em;
  line-height: 5em;
}
