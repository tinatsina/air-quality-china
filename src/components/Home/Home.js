import { Link } from 'react-router-dom';

const Home = () => (
  <>
    <ul className="home-header">
      <li>Most Polluted</li>
    </ul>
    <Link to="Country">Country Page</Link>
  </>
);

export default Home;
