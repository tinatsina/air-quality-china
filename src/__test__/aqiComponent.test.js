import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Home, { HomeList } from '../components/Home/Home';
import Country from '../components/Country/Country';
import cityMockData, { ShanghaiData } from '../__mocks__/aqiMockst';

describe('Component Snapshot Test', () => {
  test('Test Home page snap', () => {
    const snapDragon = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    const tree = snapDragon.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Test City Page Snap', () => {
    const snapCity = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Country />
        </BrowserRouter>
      </Provider>,
    );
    const tree = snapCity.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('City Page List', () => {
    const snapCity = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          {cityMockData.map((city) => (
            <HomeList key={city.id} aqi={city.aqi} city={city.city} />
          ))}
        </BrowserRouter>
      </Provider>,
    );
    const tree = snapCity.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// Testing for API fetches and calls
global.fetch = jest.fn(() => Promise.resolve({
  json: () => Promise.resolve(ShanghaiData),
}));

// Clearing the mock fetches
beforeEach(() => {
  fetch.mockClear();
});

// Testing the fetch call structure
it('Checks whether the data has been called or not', async () => {
  // Creating the variables needed for testing
  const AQI_VALUE = await ShanghaiData.data.aqi;
  const API_FETCH_SUCCESS = await ShanghaiData.status;

  expect(API_FETCH_SUCCESS).toBe('ok');
  expect(AQI_VALUE).toBe(52);
});
