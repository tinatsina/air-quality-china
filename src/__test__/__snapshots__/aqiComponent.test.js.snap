// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Snapshot Test City Page List 1`] = `
Array [
  <li
    className="home-card"
  >
    <div
      className="image-arrow-conatiner"
    >
      <img
        alt="city mapping"
        src="Asset 1.svg"
      />
      <a
        href="/Beijing"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <p
      className="city-name"
    >
      Beijing
    </p>
    <p
      className="city-aqi"
    >
      182
    </p>
  </li>,
  <li
    className="home-card"
  >
    <div
      className="image-arrow-conatiner"
    >
      <img
        alt="city mapping"
        src="Asset 1.svg"
      />
      <a
        href="/Nanjing"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <p
      className="city-name"
    >
      Nanjing
    </p>
    <p
      className="city-aqi"
    >
      52
    </p>
  </li>,
  <li
    className="home-card"
  >
    <div
      className="image-arrow-conatiner"
    >
      <img
        alt="city mapping"
        src="Asset 1.svg"
      />
      <a
        href="/Shanghai"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <p
      className="city-name"
    >
      Shanghai
    </p>
    <p
      className="city-aqi"
    >
      72
    </p>
  </li>,
  <li
    className="home-card"
  >
    <div
      className="image-arrow-conatiner"
    >
      <img
        alt="city mapping"
        src="Asset 1.svg"
      />
      <a
        href="/Tianjin"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <p
      className="city-name"
    >
      Tianjin
    </p>
    <p
      className="city-aqi"
    >
      199
    </p>
  </li>,
  <li
    className="home-card"
  >
    <div
      className="image-arrow-conatiner"
    >
      <img
        alt="city mapping"
        src="Asset 1.svg"
      />
      <a
        href="/Hangzhou"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <p
      className="city-name"
    >
      Hangzhou
    </p>
    <p
      className="city-aqi"
    >
      63
    </p>
  </li>,
  <li
    className="home-card"
  >
    <div
      className="image-arrow-conatiner"
    >
      <img
        alt="city mapping"
        src="Asset 1.svg"
      />
      <a
        href="/Shantou"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <p
      className="city-name"
    >
      Shantou
    </p>
    <p
      className="city-aqi"
    >
      55
    </p>
  </li>,
  <li
    className="home-card"
  >
    <div
      className="image-arrow-conatiner"
    >
      <img
        alt="city mapping"
        src="Asset 1.svg"
      />
      <a
        href="/Changzhou"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <p
      className="city-name"
    >
      Changzhou
    </p>
    <p
      className="city-aqi"
    >
      74
    </p>
  </li>,
  <li
    className="home-card"
  >
    <div
      className="image-arrow-conatiner"
    >
      <img
        alt="city mapping"
        src="Asset 1.svg"
      />
      <a
        href="/Jinan"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <p
      className="city-name"
    >
      Jinan
    </p>
    <p
      className="city-aqi"
    >
      127
    </p>
  </li>,
  <li
    className="home-card"
  >
    <div
      className="image-arrow-conatiner"
    >
      <img
        alt="city mapping"
        src="Asset 1.svg"
      />
      <a
        href="/Shenzhen"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <p
      className="city-name"
    >
      Shenzhen
    </p>
    <p
      className="city-aqi"
    >
      97
    </p>
  </li>,
  <li
    className="home-card"
  >
    <div
      className="image-arrow-conatiner"
    >
      <img
        alt="city mapping"
        src="Asset 1.svg"
      />
      <a
        href="/Dongli"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
    <p
      className="city-name"
    >
      Dongli
    </p>
    <p
      className="city-aqi"
    >
      165
    </p>
  </li>,
]
`;

exports[`Component Snapshot Test Test City Page Snap 1`] = `
Array [
  <nav
    className="aqi-navbar"
  >
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M328 112L184 256l144 144"
            fill="none"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="48"
          />
        </svg>
      </a>
    </div>
    <p>
      AQI DATA
    </p>
  </nav>,
  <div
    className="aqi-big-display"
  >
    <div>
      <h2 />
      <h3 />
    </div>
    <img
      alt="Map of the city"
      src="Asset 2.svg"
    />
  </div>,
  <div
    className="night-blue-list"
  >
    <p>
      CITY AIR QUALITY STATS
    </p>
  </div>,
  <ul
    className="city-header"
  >
    <li>
      <p>
        CO
      </p>
      <p />
    </li>
    <li>
      <p>
        H
      </p>
      <p />
    </li>
    <li>
      <p>
        NO2
      </p>
      <p />
    </li>
    <li>
      <p>
        O3
      </p>
      <p />
    </li>
    <li>
      <p>
        P
      </p>
      <p />
    </li>
    <li>
      <p>
        PM10
      </p>
      <p />
    </li>
    <li>
      <p>
        PM25
      </p>
      <p />
    </li>
    <li>
      <p>
        SO2
      </p>
      <p />
    </li>
    <li>
      <p>
        T
      </p>
      <p />
    </li>
    <li>
      <p>
        W
      </p>
      <p />
    </li>
  </ul>,
  <footer>
    <p>
      Original design idea by
    </p>
  </footer>,
]
`;

exports[`Component Snapshot Test Test Home page snap 1`] = `
Array [
  <nav>
    <p>
      Highest Polluters
    </p>
  </nav>,
  <div
    className="china-display"
  >
    <img
      alt="Map of China"
      src="China.svg"
    />
    <h3>
      CHINA CITIES
    </h3>
  </div>,
  <div
    className="night-blue-list"
  >
    <p>
      STATS BY CITIES
    </p>
  </div>,
  <ul
    className="home-container"
  />,
  <footer>
    <p>
      Original design idea by
      <a
        href="https://www.behance.net/sakwadesignstudio"
      >
         Nelson Sakwa on Behance.
      </a>
    </p>
  </footer>,
]
`;
